docs/initializing.rst

```diff
@@ -555,6 +555,36 @@ point into the system. It will handle the setup and initialization of the config
 and start up a Toolkit session once all the required pieces have been initialized and set up.
 
 
+Distributing application code from your site
+============================================
+
+When it's not convenient to provide `git` or `AppStore` access to your users, it might be simpler to
```

Comment:
I think we should be more forceful here - saying _When it's not convenient to provide `git` or `AppStore` access to your users_ implies that the reader understands these different approaches and pros and cons etc. (app store access not even being an option for your own tools). I think we should just tell them - do it this way! here are steps 1,2,3,4...

I think we should provide more of a step-by-step kind of messaging (with a diagram, as suggested in chat threat ) - where we say something like - if you have own apps, engines and frameworks that you wish to distribute as part of your distributed config, we suggest using the following pattern:

`[figure]`

- create custom entity
- create a custom hook by uncommenting our stuff
- (link to?) script/code to analyze a given config and upload all dependencies to the custom entity. Or handle this manually

Screenshot of what the recommended shotgun structure should look like:

![image](https://user-images.githubusercontent.com/337710/45620590-57190e80-ba75-11e8-88c6-84506a7a38d3.png)



The briefly describe the workflow: this means you can develop in git etc, and when you are about to release your config, before you upload it to shotgun to deploy it, analyze its dependncies, upload them to sg. Then at runtime, the hook will always check your shotgun cache first, before checking the actual descriptor (git, app store etc).